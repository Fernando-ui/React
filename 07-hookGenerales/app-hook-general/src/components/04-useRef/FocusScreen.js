import React, { useRef } from 'react'
import '../02-useEffect/effects.css'


export const FocusScreen = () => {


    const inputRef = useRef();
    console.log(inputRef);
    
    const handleClick = () => {
        
        // document.querySelector('input').select();
        inputRef.current.select();
        console.log(inputRef.current);
        
        
    }
    
    return (
        <>
        <div id="containerName">
        <h1>Focus Screen</h1>
        <hr/>

        <input  
        className="form-control"
        placeholder="Nombre"
        ref= { inputRef }
        />

        <button 
        className="btn btn-primary"
        id="botonFocus"
        onClick={ handleClick }
        >Focus</button>
        </div>
            
        </>
    )
}
