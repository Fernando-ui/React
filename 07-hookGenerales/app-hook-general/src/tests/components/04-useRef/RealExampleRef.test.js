import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";
import "@testing-library/jest-dom";


// jest.mock('../../../components/04-useRef/RealExampleRef');

describe("Pruebas en <RealExampleRef/>", () => {

    const wrapper = shallow(<RealExampleRef />);
    
    
    test("Debe mostrarse correctamente ", () => {
        
        expect(wrapper).toMatchSnapshot();
        
        
    });
    
    test("Debe de mostrar el componente <MultipleCustomHooks />", () => {
        
        const wrapper = shallow(<RealExampleRef />);
        wrapper.find('button').simulate('click');
        
        // wrapper.find('.btn').simulate();
        console.log(wrapper.exists());

        expect(wrapper.find('MultipleCustomHooks').exists()).toEqual(true);

    });



});
