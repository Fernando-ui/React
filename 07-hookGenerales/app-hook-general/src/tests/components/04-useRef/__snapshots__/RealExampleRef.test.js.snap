// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef/> Debe mostrarse correctamente  1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</div>
`;
